ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/managed-identity/deprovision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-managed-identity-deprovision-chain.yaml

```diff
@@ -7,6 +7,7 @@ chain:
   - chain: gather
   - ref: ipi-deprovision-deprovision-in-bastion
   - ref: azure-deprovision-role-assignment
+  - ref: azure-deprovision-sp-and-custom-role
```

Comment:
I guess this step should be moved behind `ipi-conf-azure-oidc-creds-deprovision`, so that when `ENABLE_MIN_PERMISSION_FOR_STS` enabled, it can use correct sp to run ccoctl destroy commands.

Reply:
To be safe, how about introduce `ENABLE_MIN_PERMISSION_FOR_STS` ENV for `ipi-conf-azure-oidc-creds-provision` and `ipi-conf-azure-oidc-creds-deprovision`, then judge the ENV instead of `azure_minimal_permission_sts`, so that we can catch the issue easily.

Reply:
We have check [file existence](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/azure/oidc-creds-provision/ipi-conf-azure-oidc-creds-provision-commands.sh#L18-L26) in ipi-conf-azure-oidc-creds-provision and ipi-conf-azure-oidc-creds-deprovision.
Do you think that we still need to have `ENABLE_MIN_PERMISSION_FOR_STS ENV ` to control that?

Reply:
How about [this](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/aws/oidc-creds-provision/ipi-conf-aws-oidc-creds-provision-commands.sh#L7-L18), once that, when we put the steps in a wrong place in a chain, then it would error out. If we follow the current codes, it would use `${CLUSTER_PROFILE_DIR}/osServicePrincipal.json` to move forwards, not easy to find out the issue.

Reply:
got it, updated.